api/users/serializers.py

```diff
@@ -40,8 +73,11 @@ def absolute_url(self, obj):
         return obj.absolute_url
 
     def update(self, instance, validated_data):
-        # TODO
-        pass
+        assert isinstance(instance, User), 'instance must be a User'
+        for attr, value in validated_data.items():
```

Comment:
So far it is working in functional tests and my own testing.  My understanding is that Q is using the . notation to read and write from the fields.  gihHub is translated to social.github via the "source=" option in the serializer.